iwyu_tool.py

```diff
@@ -197,6 +214,30 @@ def slice_compilation_db(compilation_db, selection):
 
     return new_db
 
+def process_invocations(invocations, verbose, formatter, jobs):
+    if jobs == 1:
+        for invocation in invocations:
+            print(formatter(invocation.run_iwyu(verbose).get()))
+        return
+
+    pending = []
+    while invocations or pending:
+        # get complete iwyu processes
+        complete = [proc for proc in pending if proc.poll()]
+
+        for proc in complete:
+            # handle iwyu output
+            pending.remove(proc)
+            print(formatter(proc.get()))
+            proc.close()
+
+        n = jobs - len(pending)
+        pending.extend(e.run_iwyu(verbose) for e in invocations[:n])
+        invocations = invocations[n:]
+
+        from time import sleep
+        sleep(0) # yield cpu
```

Comment:
In my testing, I haven't seen `sleep(0)` yield -- the Python process would spin like crazy polling the pending processes. I'll re-run my tests with this patch to make sure I wasn't seeing something else. But I think you need to increase this sleep to yield effectively.

Reply:
Can't reproduce now, this seems to work as expected.

Reply:
Ah, this was because the special-casing for `jobs == 1` never executed the process pool. When I remove that, I'm back to 100% CPU for python.